handlers/inventory_nodes.go

```diff
@@ -18,88 +18,191 @@ package handlers
+	node, err := s.Nodes.Add(ctx, models.RemoteNodeType, req.Name, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &api.AddRemoteNodeResponse{
+		Remote: node.(*api.RemoteNode),
+	}
+	return res, nil
 }
 
-// func (s *NodesServer) List(ctx context.Context, req *inventory.NodesListRequest) (*inventory.NodesListResponse, error) {
-// 	logger.Get(ctx).Infof("%#v", req)
-// 	return &inventory.NodesListResponse{
-// 		Node: []*inventory.Node{
-// 			{
-// 				Id: 1,
-// 			},
-// 		},
-// 		BareMetal: []*inventory.BareMetalNode{
-// 			{
-// 				Id: 2,
-// 			},
-// 		},
-// 		Container: []*inventory.ContainerNode{
-// 			{
-// 				Id: 3,
-// 			},
-// 		},
-// 	}, nil
-// }
-
-// func (s *NodesServer) Get(ctx context.Context, req *inventory.NodesGetRequest) (*inventory.NodesGetResponse, error) {
-// 	logger.Get(ctx).Infof("%#v", req)
-// 	return &inventory.NodesGetResponse{
-// 		OnlyOne: &inventory.NodesGetResponse_BareMetal{
-// 			BareMetal: &inventory.BareMetalNode{
-// 				Id: 1,
-// 			},
-// 		},
-// 	}, nil
-// }
-
-// func (s *NodesServer) AddBareMetal(ctx context.Context, req *inventory.AddBareMetalRequest) (*inventory.AddBareMetalResponse, error) {
-// 	node := s.Store.AddBareMetalNode(req)
-// 	return &inventory.AddBareMetalResponse{
-// 		Node: node,
-// 	}, nil
-// }
-
-// func (s *NodesServer) AddMySQLdExporter(ctx context.Context, req *inventory.AddMySQLdExporterRequest) (*inventory.AddMySQLdExporterResponse, error) {
-// 	exporter := s.Store.AddMySQLdExporter(req)
-// 	return &inventory.AddMySQLdExporterResponse{
-// 		Agent: exporter,
-// 	}, nil
-// }
+// AddRDSNode adds AWS RDS Node.
+func (s *NodesServer) AddRDSNode(ctx context.Context, req *api.AddRDSNodeRequest) (*api.AddRDSNodeResponse, error) {
+	node, err := s.Nodes.Add(ctx, models.RemoteNodeType, req.Name, &req.Hostname, &req.Region)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &api.AddRDSNodeResponse{
+		Rds: node.(*api.RDSNode),
+	}
+	return res, nil
+}
+
+// ChangeBareMetalNode changes bare metal Node.
+func (s *NodesServer) ChangeBareMetalNode(ctx context.Context, req *api.ChangeBareMetalNodeRequest) (*api.ChangeBareMetalNodeResponse, error) {
+	if err := s.Nodes.Change(ctx, req.Id, req.Name); err != nil {
+		return nil, err
+	}
+
+	return new(api.ChangeBareMetalNodeResponse), nil
+}
+
+// ChangeVirtualMachineNode changes virtual machine Node.
+func (s *NodesServer) ChangeVirtualMachineNode(ctx context.Context, req *api.ChangeVirtualMachineNodeRequest) (*api.ChangeVirtualMachineNodeResponse, error) {
+	if err := s.Nodes.Change(ctx, req.Id, req.Name); err != nil {
+		return nil, err
+	}
+
+	return new(api.ChangeVirtualMachineNodeResponse), nil
+}
+
+// ChangeContainerNode changes container Node.
+func (s *NodesServer) ChangeContainerNode(ctx context.Context, req *api.ChangeContainerNodeRequest) (*api.ChangeContainerNodeResponse, error) {
```

Comment:
Maybe it should be Rename instead of Change?

Reply:
We may add hostname and some other fields in the future.